src/snowflake/cli/api/config.py

```diff
@@ -286,8 +286,12 @@ def get_config_value(*path, key: str, default: Optional[Any] = Empty) -> Any:
         raise
 
 
-def get_config_bool_value(*path, key: str, default: Optional[Any] = Empty) -> bool:
-    value = get_config_value(*path, key=key, default=default)
+def get_config_bool_value(*path, key: str, default: Any = Empty) -> bool | None:
```

Comment:
that signature doesn't make sense to me, looks like we can return `Optional[Any]` with the current logic. That wasn't the case before because we tried to cast any value to a bool.

Do we really need `Any` anyway? `Optional[bool]` is probably a lot more intuitive.